src/factory.ts

```diff
@@ -464,10 +464,11 @@ export class Factory {
     return fingering;
   }
 
-  StringNumber(params: { number: string; position: string }): StringNumber {
+  StringNumber(params: { number: string; position: string; radius?: number }): StringNumber {
     const stringNumber = new StringNumber(params.number);
     stringNumber.setPosition(params.position);
     stringNumber.setContext(this.context);
+    if (params.radius != undefined) stringNumber.radius = params.radius;
```

Comment:
Not a fan of manipulating the instance outside an API call.

Reply:
Me neither but `radius` has no setter / getter and it is a public member. OSMD sets radius to 0 in order to present stringNumbers without circles.

What do you think about my considerations above?:

> I wonder if it really makes sense to allow to set different radius:
> 
> - Radii bigger than 8 overlap on chords and smaller than 7 overlap with the number.
> - the this.setWidth(20); // ??? is also somehow related to the radius.
> 
> I would recommend changing the class to allow the configuration of wheter a circle should be drawn.

With this I mean amking radius protected and adding a function or an option to say if the circle should be drawn.

Reply:
New approach pushed.